torchtext/experimental/vocab.py

```diff
@@ -197,6 +198,34 @@ def __getitem__(self, token: str) -> int:
         """
         return self.vocab[token]
 
+    @torch.jit.export
+    def set_default_index(self, index: int) -> None:
+        r"""
+        Args:
+            index: Value of default index. This index will be returned when OOV token is queried
+        """
+        self.vocab.set_default_index(index)
+
+    @torch.jit.export
+    def get_default_index(self) -> Optional[int]:
+        r"""
+        Returns:
+            index (optional[int]): Value of default index if it is set.
+        """
+        return self.vocab.get_default_index()
+
+    @torch.jit.export
+    def reassign_token(self, token: str, index: int) -> None:
```

Comment:
Well actually both will affect the indices of other tokens (except for insertion when it is at the end)

```
v1 = Vocab(['a','b','c'])
v1.reassign_token('b',0)
v1['a'] is 1 #index of a changes
```

```
v1 = Vocab(['a','b','c'])
v1.insert_token('d',0)
v1['a'] is 1 #index of 'a','b','c' changes
```

The only difference being is in case of reassignment we expect the token is already present and in case of insertion, the token is not present. By handling them together, we could simplify the interface and we never throw error any more due to presence/absence of token as is the case in present form. Also by definition `__setitem__` supports both changing the values of keys as well as adding new keys. Based on this, I feel handling them together  should be a more natural choice for user :)